xls/contrib/mlir/IR/xls_ops.cc

```diff
@@ -390,6 +390,28 @@ Type getI1TypeOf(Type type) {
 
 namespace mlir::xls {
 
+LogicalResult LiteralOp::verifyRegions() {
+  Block& b = getInitializerBlock();
+  // Verify return op & typing:
+  auto ret = cast<YieldOp>(b.getTerminator());
+  if (ret->operand_type_begin() == ret.operand_type_end()) {
+    return emitOpError("initializer region cannot return void");
+  }
+  if (*(ret->operand_type_begin()) != getType()) {
```

Comment:
you're assuming only one result here but you haven't checked it (only that there are > 0 results).

Also can you just use if (ret.getOperand().getType() != getType()) ?

Reply:
sorry, it's a YieldOp so it may have multiple operands, so you need ret.getOperand(0).